resources/styles/molecules/tabs/BpTabs.vue

```diff
@@ -96,117 +100,143 @@ export default {
-        isFocusedTab (tab) {
-            return this.currentFocusTab === tab
-        },
-        navigateNextTab (evt) {
-            if (this.nextTabKey) {
-                evt.preventDefault();
-                const nextTabButton = this.$el.querySelector(`[id="${this.nextTabKey}"]`)
-                nextTabButton.focus()
-            }
-        },
-        navigatePrevTab (evt) {
-            if (this.prevTabKey) {
-                evt.preventDefault();
-                const prevTabButton = this.$el.querySelector(`[id="${this.prevTabKey}"]`)
-                prevTabButton.focus()
+        }
+    })
+
+    const wrapComp = computed({
+        get: () => {
+            if (!props.wrap) {
+                return !props.vertical
+            } else {
+                return props.wrap
             }
-        },
-        attachKeyboardListeners () {
-            const firstTabEl = this.$el.querySelector(`[id="${this.firstTabKey}"]`)
-            const lastTabEl = this.$el.querySelector(`[id="${this.lastTabKey}"]`)
+        }
+    })
 
-            this.$el.addEventListener('keydown', event => {
-                if (event.keyCode === 36) {
-                    event.preventDefault()
+    const tabKeys = computed({
+        get: () => Object.keys(props.tabs)
+    })
 
-                    firstTabEl.focus()
-                    this.activeTab = this.firstTabKey
-                }
+    const firstTabKey = computed({
+        get: () => tabKeys.value[0]
+    })
 
-                if (event.keyCode === 35) {
-                    event.preventDefault()
+    const lastTabKey = computed({
+        get: () => tabKeys.value[tabKeys.value.length - 1]
+    })
 
-                    lastTabEl.focus()
-                    this.activeTab = this.lastTabKey
-                }
-            })
-        },
-    },
+    const currentFocusTabIndex = computed({
+        get: () => tabKeys.value.indexOf(currentFocusTab.value)
+    })
+
+    const nextTabKey = computed({
+        get: () => {
+            return currentFocusTab.value === lastTabKey.value
+                ? (wrapComp.value ? firstTabKey.value : null)
+                : tabKeys.value[currentFocusTabIndex.value + 1]
+        }
+    })
+
+    const prevTabKey = computed({
+        get: () => {
+            return currentFocusTabIndex.value === 0
+                ? (wrapComp.value ? lastTabKey.value : null)
+                : tabKeys.value[currentFocusTabIndex.value - 1]
+        }
+    })
+
+    onMounted(() => {
+        setInitFocusTab()
+        setInitActiveTab()
+        attachKeyboardListeners()
+    })
+
+    function setInitActiveTab () {
+        if (location.hash && tabKeys.indexOf(location.hash.substr(1)) !== -1) {
+            activeTab.value = location.hash.substr(1)
+        } else if (initialTabComp.value) {
+            activeTab.value = initialTabComp.value
+        } else {
+            [activeTab.value] = props.tabs
+        }
+    }
+        
+    function setFocusTab (tab) {
```

Comment:
It's not having any impact on the user, but because when you click on a button it also creates a focus event, each `@click` event is calling both `selectTab` and `setFocusTab`, which means `selectTab` is called twice every time. Just a note, and this was an issue with the original Vue 2 implementation as well.